Signal/src/ViewControllers/NewGroupViewController.m

```diff
@@ -66,243 +79,353 @@ - (instancetype)initWithCoder:(NSCoder *)aDecoder
+    [_tableViewController.view autoPinEdgeToSuperviewEdge:ALEdgeBottom];
 
-    [self.tableView reloadData];
+    [self updateTableContents];
+}
 
-    // Restore selection state.
-    for (Contact *contact in selectedContacts) {
-        if ([contacts containsObject:contact]) {
-            NSInteger row = (NSInteger)[contacts indexOfObject:contact];
-            [self.tableView selectRowAtIndexPath:[NSIndexPath indexPathForRow:row inSection:0]
-                                        animated:NO
-                                  scrollPosition:UITableViewScrollPositionNone];
-        }
+- (UIView *)firstSectionHeader
+{
+    UIView *firstSectionHeader = [UIView new];
+    firstSectionHeader.backgroundColor = [UIColor whiteColor];
+    UIView *threadInfoView = [UIView new];
+    [firstSectionHeader addSubview:threadInfoView];
+    [threadInfoView autoPinWidthToSuperviewWithMargin:16.f];
+    [threadInfoView autoPinHeightToSuperviewWithMargin:16.f];
+
+    const CGFloat kAvatarSize = 68.f;
+    UIImageView *avatarView = [UIImageView new];
+    _avatarView = avatarView;
+    avatarView.layer.borderColor = UIColor.clearColor.CGColor;
+    avatarView.layer.masksToBounds = YES;
+    avatarView.layer.cornerRadius = kAvatarSize / 2.0f;
+    avatarView.contentMode = UIViewContentModeScaleAspectFill;
+    [threadInfoView addSubview:avatarView];
+    [avatarView autoVCenterInSuperview];
+    [avatarView autoPinEdgeToSuperviewEdge:ALEdgeLeft];
+    [avatarView autoSetDimension:ALDimensionWidth toSize:kAvatarSize];
+    [avatarView autoSetDimension:ALDimensionHeight toSize:kAvatarSize];
+    [self updateAvatarView];
+
+    UITextField *groupNameTextField = [UITextField new];
+    _groupNameTextField = groupNameTextField;
+    groupNameTextField.textColor = [UIColor blackColor];
+    groupNameTextField.font = [UIFont ows_dynamicTypeTitle2Font];
+    groupNameTextField.placeholder = NSLocalizedString(@"NEW_GROUP_NAMEGROUP_REQUEST_DEFAULT", @"");
+    groupNameTextField.delegate = self;
+    [groupNameTextField addTarget:self
+                           action:@selector(groupNameDidChange:)
+                 forControlEvents:UIControlEventEditingChanged];
+    [threadInfoView addSubview:groupNameTextField];
+    [groupNameTextField autoVCenterInSuperview];
+    [groupNameTextField autoPinEdge:ALEdgeLeft toEdge:ALEdgeRight ofView:avatarView withOffset:16.f];
+    [groupNameTextField autoPinEdgeToSuperviewEdge:ALEdgeRight withInset:16.f];
+
+    [avatarView addGestureRecognizer:[[OWSAnyTouchGestureRecognizer alloc] initWithTarget:self
+                                                                                   action:@selector(avatarTouched:)]];
+    avatarView.userInteractionEnabled = YES;
+
+    return firstSectionHeader;
+}
+
+- (void)avatarTouched:(UIGestureRecognizer *)sender
+{
+    if (sender.state == UIGestureRecognizerStateRecognized) {
+        [self showChangeGroupAvatarUI:nil];
```

Comment:
nit: please label your params, otherwise I have to go to the caller to have a clue what this `nil` is about.

e.g.  `showChangeGroupAvatarUIWithSender:` or just drop the param since we're not using it.